src/components/Reports/StaffExpenseReport/DownloadButtonGroup/downloadReport.test.tsx

```diff
@@ -0,0 +1,157 @@
+      mockT,
+      mockLocale,
+    );
+
+    expect(setAttributeMock).toHaveBeenCalledWith(
+      'href',
+      expect.stringContaining('data:text/csv'),
+    );
+
+    expect(setAttributeMock).toHaveBeenCalledWith(
+      'download',
+      expect.stringContaining(`Income Report.csv`),
+    );
+    expect(appendChildMock).toHaveBeenCalledWith(realLink);
+    expect(clickMock).toHaveBeenCalled();
+    expect(removeChildMock).toHaveBeenCalledWith(realLink);
+    expect(enqueueSnackbarMock).not.toHaveBeenCalled();
+  });
+
+  it('downloads an expense report when ReportType.Expense is passed as an argument', () => {
+    const realLink = document.createElement('a');
+    jest.spyOn(realLink, 'setAttribute').mockImplementation(setAttributeMock);
+    jest.spyOn(realLink, 'click').mockImplementation(clickMock);
+
+    jest.spyOn(document, 'createElement').mockReturnValue(realLink);
+
+    const expenseMockData = mockData.filter(
+      (transaction) => transaction.total < 0,
+    );
+    downloadCsv(
+      ReportType.Expense,
+      enqueueSnackbarMock,
+      expenseMockData,
+      mockT,
+      mockLocale,
+    );
+
+    expect(setAttributeMock).toHaveBeenCalledWith(
+      'href',
+      expect.stringContaining('data:text/csv'),
+    );
+
+    expect(setAttributeMock).toHaveBeenCalledWith(
+      'download',
+      expect.stringContaining(`Expense Report.csv`),
+    );
+    expect(appendChildMock).toHaveBeenCalledWith(realLink);
+    expect(clickMock).toHaveBeenCalled();
+    expect(removeChildMock).toHaveBeenCalledWith(realLink);
+    expect(enqueueSnackbarMock).not.toHaveBeenCalled();
+  });
+
+  it('downloads a combined report when ReportType.Combined is passed as an argument', () => {
+    const realLink = document.createElement('a');
+    jest.spyOn(realLink, 'setAttribute').mockImplementation(setAttributeMock);
+    jest.spyOn(realLink, 'click').mockImplementation(clickMock);
+
+    jest.spyOn(document, 'createElement').mockReturnValue(realLink);
+
+    downloadCsv(
+      ReportType.Combined,
+      enqueueSnackbarMock,
+      mockData,
+      mockT,
+      mockLocale,
+    );
+
+    expect(setAttributeMock).toHaveBeenCalledWith(
+      'href',
+      expect.stringContaining('data:text/csv'),
+    );
+
+    expect(setAttributeMock).toHaveBeenCalledWith(
+      'download',
+      expect.stringContaining(`Combined Report.csv`),
+    );
+    expect(appendChildMock).toHaveBeenCalledWith(realLink);
+    expect(clickMock).toHaveBeenCalled();
+    expect(removeChildMock).toHaveBeenCalledWith(realLink);
+    expect(enqueueSnackbarMock).not.toHaveBeenCalled();
+  });
+
+  it('shows snackbar and returns when no transactions are provided', () => {
+    downloadCsv(ReportType.Income, enqueueSnackbarMock, [], mockT, mockLocale);
+
+    expect(enqueueSnackbarMock).toHaveBeenCalledWith(
+      'No transactions to download',
+      { variant: 'error' },
+    );
+  });
+
+  it('returns early if an invalid ReportType is provided', () => {
```

Comment:
I think we can remove this test. I'm good with trusting TypeScript to enforce that a valid enum value is passed as the report type.